### PR TITLE
CTCP-4759 : Implementing sequential navigation in NormalMode

### PR DESCRIPTION
**Pull request by jackheslop96**

### CODE REVIEW

File: app/models/journeyDomain/package.scala [hunk 1]
```diff
@@ -89,52 +102,141 @@ package object domain {
...
               }
           }
         }
       }
       UserAnswersReader(fn)
     }
   }
+
+  type Pages   = Seq[Page]
+  type Read[T] = Pages => UserAnswersReader[T]
```
Comment (joecefnhedog) [inline app/models/journeyDomain/package.scala:150]: I think this confused me initially, I misstook it to be the play.json.lin.Read thing

File: app/pages/sections/locationOfGoods/LocationOfGoodsSection.scala [hunk 2]
```diff
@@ -16,13 +16,18 @@
...
+import play.api.mvc.Call
 
 case object LocationOfGoodsSection extends Section[JsObject] {
 
   override def path: JsPath = RouteDetailsSection.path \ toString
 
   override def toString: String = "locationOfGoods"
+
+  override def route(userAnswers: UserAnswers, mode: Mode): Option[Call] =
+    Some(routes.CheckYourAnswersController.onPageLoad(userAnswers.lrn, mode))
```
Comment (joecefnhedog) [inline app/pages/sections/locationOfGoods/LocationOfGoodsSection.scala:32]: I wasn't sure if this was supposed to be CYA

**Review by joecefnhedog — COMMENTED** [review]
This looks good to me

File: app/models/journeyDomain/RouteDetailsDomain.scala [hunk 3]
```diff
@@ -45,61 +45,74 @@ case class RouteDetailsDomain(
...
-      loadingAndUnloading           <- UserAnswersReader[LoadingAndUnloadingDomain]
-    } yield RouteDetailsDomain(
-      specificCircumstanceIndicator,
-      routing,
-      transit,
-      exit,
-      locationOfGoods,
-      loadingAndUnloading
-    )
+    specificCircumstanceIndicatorReader(Nil).flatMap {
```
[the remark this answers is not in the document]
Comment (jackheslop96) [reply app/models/journeyDomain/RouteDetailsDomain.scala]: Updated to make use of the `aqpply` and `map` methods - let me know what you think

File: app/models/journeyDomain/package.scala [hunk 4]
```diff
@@ -89,52 +102,141 @@ package object domain {
...
+        case x                      => pages :+ x
+      }
+  }
+
+  implicit class RichRead[A](value: Read[A]) {
+
+    def map[T <: JourneyDomainModel](f: A => T): Read[T] =
+      apply {
+        a =>
+          val t = f(a)
```
[the remark this answers is not in the document]
Comment (jackheslop96) [reply app/models/journeyDomain/package.scala]: `t` used twice

File: app/pages/sections/locationOfGoods/LocationOfGoodsSection.scala (hunk 2 above)
Comment (jackheslop96) [reply app/pages/sections/locationOfGoods/LocationOfGoodsSection.scala:32]: Yep this is fine it's pointing to the location of goods CYA

**Review by joecefnhedog — APPROVED** [review]

**Pull request merged**